File: trash/app/src/main/java/com/rikkei/training/fragment/base/BaseFragment.java
package com.rikkei.training.fragment.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public abstract class BaseFragment<AC extends AppCompatActivity> extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(getLayoutId(),container,false);
        return v;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initView();
        initData();
    }

    protected abstract void initData();

    protected abstract void initView();

    protected abstract int getLayoutId();

    protected <v extends View> v findViewById(@IdRes int resId){
        return getActivity().findViewById(resId);
    }

    protected AC getParentActivity(){
        return (AC)getActivity();
    }

}

File: FragmentKotlin/app/src/main/java/com/rikkei/training/fragmentkotlin/base/BaseFragment.kt
package com.rikkei.training.fragmentkotlin.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

abstract class  BaseFragment<AC : AppCompatActivity> : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        var v:View =inflater.inflate(getLayoutId(),container,false);
        return v
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initData()
    }

    abstract fun initData()

    protected fun getParentActivity():AC{
        return activity as AC
    }

    abstract fun getLayoutId(): Int
}